libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -81,6 +81,12 @@ func New(cfg *common.Config) (processors.Processor, error) {
 
 // Run enriches the given event with the host meta data
 func (p *addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
+	// If host fields exist(besides host.name added by libbeat) in event, skip add_host_metadata.
+	hostFields, _ := event.Fields.GetValue("host")
+	if hostFields != nil && len(hostFields.(common.MapStr)) > 1 {
```

Comment:
Maybe to do a more intrusive check for `host.name` instead of just checking the length of the map?

Reply:
@ChrsMark Good point! I changed it to remove host.name field first and then check the length to see if there are other fields exist.

Reply:
Ah, but it still can be that `len(hostFields)==1` but it only contains `host.something`. Shouldn't the "delete" of `host.name` happen before any checks 🤔 ?

Reply:
Yeah I didn't consider that originally because with libbeat adding `host.name`, it will always have a `host.name` field. But since we might change when libbeat to add `host.name`, I'm changing the code to below for checking both conditions with or without `host.name`(unit test is also added for it):
```
         hostFields, err := event.Fields.GetValue("host")
	fmt.Println("hostFields = ", hostFields)
	if err == nil && hostFields != nil && len(hostFields.(common.MapStr)) >= 1 {
		if hasName, _ := hostFields.(common.MapStr).HasKey("name"); hasName {
			// other host fields exist on top of host.name, skip add_host_metadata.
			if len(hostFields.(common.MapStr)) > 1 {
				return event, nil
			}
		} else {
			// host.name does not exist but other host fields exist, skip add_host_metadata.
			if len(hostFields.(common.MapStr)) > 0 {
				return event, nil
			}
		}
	}
```

Reply:
👍